src/python/WMCore/DataStructs/Run.py

```diff
@@ -49,13 +49,24 @@ def __ne__(self, rhs):
 
     def __lt__(self, rhs):
         """
-        Compare on run # first, then by lumis as a list is compared
+        Compare on run # first, then by lumis as a list, then by events in each lumi
         """
+        # check run number
         if self.run != rhs.run:
             return self.run < rhs.run
+        # if same run number, check list of lumi sections
         if sorted(self.eventsPerLumi.keys()) != sorted(rhs.eventsPerLumi.keys()):
             return sorted(self.eventsPerLumi.keys()) < sorted(rhs.eventsPerLumi.keys())
-        return self.eventsPerLumi < rhs.eventsPerLumi
+        # if same list of lumis, check events in each lumi section
+        for self_key, rhs_key in zip(sorted(self.eventsPerLumi), sorted(rhs.eventsPerLumi)):
```

Comment:
In case you decide to update the assert check, and given the discussion raised in this PR, I might have a minor suggestion:
```
for lumiNumber in sorted(self.eventsPerLumi):
    if self.eventsPerLumi[lumiNumber] == rhs.eventsPerLumi.get(lumiNumber):
        continue
    else:
        return self.eventsPerLumi[lumiNumber] < rhs.eventsPerLumi.get(lumiNumber)
```

I do not know whether it works though. Neither if it would improve the performance :-D

PS.: keep in mind we use lower camelCase for variable names (even though you will find many modules with a mix of conventions).

Reply:
This is not a minor suggestion, this is a supernice suggestion! Pushed in the last commits, squashed into the previous ones